docs/modeling-your-data/modeling-your-data-with-dbt/dbt-quickstart/utils/index.md

```diff
@@ -0,0 +1,233 @@
+
+```sql
+{{
+  config(
+    materialized='incremental'
+  )
+}}
+
+{% set quarantined_query = snowplow_utils.base_create_snowplow_quarantined_sessions() %}
+
+{{ quarantined_query }}
+```
+
+This macro doesn't accept any arguments, and simply generates a table which contains a column named `session_identifier`, containing all session identifiers of sessions that have been quarantined due to exceeding the maximum session length, to avoid long table scans.
+
+### 4. Setting up the incremental manifest macro
+Next, within the `snowplow_incremental_manifest.sql` file, you can call the `base_create_snowplow_incremental_manifest` macro as follows:
+
+```sql
+{{
+  config(
+    materialized='incremental'
+  )
+}}
+
+{% set incremental_manifest_query = snowplow_utils.base_create_snowplow_incremental_manifest() %}
+
+{{ incremental_manifest_query }}
+```
+
+Much like with the quarantined sessions macro, this does not accept any arguments and generates the boilerplate for the incremental manifest table that Snowplow leverages.
+
+### 5. Setting up the new event limits macro
+For the `snowplow_base_new_event_limits` model, you need to add a few extra macros into the mix, which you can do as follows:
+
+```sql
+
+{%- set models_in_run = snowplow_utils.base_get_enabled_snowplow_models() -%}
+
+{% set min_last_success,
+         max_last_success,
+         models_matched_from_manifest,
+         has_matched_all_models = snowplow_utils.get_incremental_manifest_status(ref('snowplow_incremental_manifest'),
+                                                                                 models_in_run) -%}
+
+
+{% set run_limits_query = snowplow_utils.get_run_limits(min_last_success,
+                                                          max_last_success,
+                                                          models_matched_from_manifest,
+                                                          has_matched_all_models,
+                                                          var("snowplow__start_date")) -%}
+
+
+{{ run_limits_query }}
+```
+
+Here there are a couple of variables that can be used to modify the `new_event_limits` table setup. Firstly, if you chose to name your `snowplow_incremental_manifest` model differently, be sure to reference that properly in the `get_incremental_manifest_status` macro call. If for example, you chose to call the manifest model `incremental_manifest` by naming your file `incremental_manifest.sql` instead of the prescribed `snowplow_incremental_manifest.sql`, then you would reflect that using the following macro call:
+
+```sql
+{% set min_last_success,
+         max_last_success,
+         models_matched_from_manifest,
+         has_matched_all_models = snowplow_utils.get_incremental_manifest_status(ref('incremental_manifest'),
+                                                                                 models_in_run) -%}
```

Comment:
Might be worth adding a comment at the top of this one to call out it's an example, not a whole new thing they have to copy and add? Should be obvious but we both know people will just read the code bits and not the text around it...